src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -68,30 +68,95 @@ public VectorTypeAttribute(params int[] dims)
     /// column encapsulates.
     /// </summary>
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
-    public sealed class ColumnAttribute : Attribute
+    public sealed class LoadColumnAttribute : Attribute
     {
-        public ColumnAttribute(string ordinal, string name = null)
+        /// <summary>
+        /// Initializes new instance of <see cref="LoadColumnAttribute"/>.
+        /// </summary>
+        /// <param name="ordinal">The index of the column in the text file.</param>
+        /// <param name="name">The optional name of the column, if it should be different from the name of the field or property where the attribute is positioned.</param>
+        /// <param name="loadAllOthers">Wheather to load all columns, besides the one specified in the ordinal.</param>
+        public LoadColumnAttribute(int ordinal, string name = null, bool loadAllOthers = false)
+             : this(ordinal.ToString(), name)
+        {
+            Sources = new List<TextLoader.Range>();
+            var range = new TextLoader.Range(ordinal);
+            range.AllOther = loadAllOthers;
+            Sources.Add(range);
+        }
+
+        /// <summary>
+        /// Initializes new instance of <see cref="LoadColumnAttribute"/>.
+        /// </summary>
+        /// <param name="start">The starting column index, for the range.</param>
+        /// <param name="end">The ending column index, for the range.</param>
+        /// <param name="name">The optional name of the column, if it should be different from the name of the field or property where the attribute is positioned.</param>
+        /// <param name="columnIndexes">Distinct text file column indces to load as part of this column.</param>
+        public LoadColumnAttribute(string start, string end = null, string name = null, int[] columnIndexes = null)
```

Comment:
>string [](start = 35, length = 6)

start and end should be integers here.

Reply:
can't have nullable ints. we can't leave 'end' as Auto, if i flip to ints. that ok?

---
In reply to: [242669241](https://github.com/dotnet/machinelearning/pull/1878#discussion_r242669241) [](ancestors = 242669241)

Reply:
If we want to preserve the '10-*' logic, I would go with a different attribute for it (`[LoadAllColumns(start: 10)]` or something like that).

---
In reply to: [242741038](https://github.com/dotnet/machinelearning/pull/1878#discussion_r242741038) [](ancestors = 242741038,242669241)

Reply:
I don't think the '*' is going to be as useful in API as it is in cmdline.

---
In reply to: [243120230](https://github.com/dotnet/machinelearning/pull/1878#discussion_r243120230) [](ancestors = 243120230,242741038,242669241)

Reply:
remove

---
In reply to: [243120291](https://github.com/dotnet/machinelearning/pull/1878#discussion_r243120291) [](ancestors = 243120291,243120230,242741038,242669241)